src/bvh/bvh.rs

```diff
@@ -197,20 +253,18 @@ impl BVHNode {
     /// [`BVH`]: struct.BVH.html
     /// [`Vec`]: https://doc.rust-lang.org/std/vec/struct.Vec.html
     ///
-    pub fn traverse_recursive(&self, ray: &Ray, indices: &mut Vec<usize>) {
+    pub fn traverse_recursive(&self, nodes: &Vec<BVHNode>, ray: &Ray, indices: &mut Vec<usize>) {
```

Comment:
Modify this function, so that it is static (no &self) and instead takes an index of the current node.

It looks like an antipattern to call a function on an element of an array and pass the array to the function.